arkouda/pdarrayclass.py

```diff
@@ -2622,13 +2622,78 @@ def rotr(x, rot) -> pdarray:
 
 
 @typechecked
-def power(pda: pdarray, pwr: Union[int, float, pdarray]) -> pdarray:
-    return pda**pwr
+def power(pda: pdarray, pwr: Union[int, float, pdarray], where: Union[bool, pdarray] = True) -> pdarray:
+    """
+    Raises an input value or array to a power, can be subjected to a condition that if True will raise value to desiered
+    power and if False will return the original value.
+
+    Notes:
+    This function was modified to more closely match the implementation of numpy's power function. Also fixes
+    an error to make a False condition return the original value, as expected.
```

Comment:
I wouldn't say it `fixes an error`, but rather deviates from numpys implementation of `where`

Reply:
I definitely agree with @pierce314159 here. That was intended `NumPy` behavior, but something that we want to operate differently.